### PR TITLE
[Integration Manager] Demandware navigation fixes

### PR DESCRIPTION
**Pull request by jnurse**
This PR updates the site navigation to work with child categories, and fixes the demandware navigation parsing to return the correct structure for the NavMenu component. 

 **JIRA**: https://mobify.atlassian.net/browse/WEB-1256
 **Linked PRs**: n/a

## Changes
- Fixed demandware navigation parser to return correct structure
- Added action/reducer to show navigation panels correctly.

## How to test-drive this PR
- Set up tag injector for the demandware sandbox (http://mobify-tech-prtnr-na03-dw.demandware.net/on/demandware.store/Sites-2017refresh-Site/default/Home-Show)
- Run `npm run dev`
- Preview [the demandware site](https://preview.mobify.com/?url=http%3A%2F%2Fmobify-tech-prtnr-na03-dw.demandware.net%2Fon%2Fdemandware.store%2FSites-2017refresh-Site%2Fdefault%2FHome-Show&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Open the site menu and try using the navigation
- You will only see a URL change when you reach a leaf in the navigation since PLP routes are being added in another PR

### CODE REVIEW

**Comment by jeremywiebe** [comment]
Navigation works, not sure if I should be able to navigate when I select a leaf entry.  I get this error, but suspect this PR might not address that. 

screen shot 2017-03-15 at 2 46 49 pm

**Comment by jnurse** [comment]
@jeremywiebe yes that's right, I mentioned it in the how to test instructions but leafs will only change the URL since PLP routes are being added in another PR.

File: web/app/containers/navigation/container.js [hunk 1]
```diff
@@ -31,15 +32,23 @@ const itemFactory = (type, props) => {
...
-        // a path, instead of a full url.
-        const routerPath = url.pathname + url.search + url.hash
-        router.push(routerPath)
-        closeNavigation()
+    const {path, isOpen, root, closeNavigation, router, setNavigationPath} = props
+
+    const onPathChange = (path, isLeaf) => {
+        if (isLeaf) {
+            let routerPath = path
+            if (!/^\//.test(path)) {
```
Comment (jeremywiebe) [inline web/app/containers/navigation/container.js]: Dumb question. What is this verifying? I _think_ it is making sure the path starts with a `/`, but I'm not sure why that's needed...

Comment (jeremywiebe) [inline web/app/containers/navigation/container.js]: This feels like something we should probably extract into a utility function then.  Maybe `extractPathFromUrl(urlOrPath)` ?

**Pull request merged**